### PR TITLE
Ansible updates

### PR DESCRIPTION
**Pull request by eparis**
Supports certs, kube-system namespace, additional vagrant providers and cleanups.

### CODE REVIEW

**Comment by eparis** [comment]
@jeffbean @jayunit100
 
Please ok your commits

File: contrib/ansible/vagrant/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+## Vagrant deployer for Kubernetes Ansible
+
+This deployer sets up a vagrant cluster and installs kubernetes with flannel on it.
+
+## Before you start !
+
+You will need to pick your vagrant provider. Currently supported are openstack, libvirt, and virtualbox.
```
Comment (jeffbean) [inline contrib/ansible/vagrant/README.md]: Can we add how to pick. Once I have picked do I have to take any actions? How do I configure that provider?

File: contrib/ansible/vagrant/openstack_config.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+os_username: eparis
```
Comment (jeffbean) [inline contrib/ansible/vagrant/openstack_config.yml]: Do you want to commit this info? Should we add to gitignore once it is an example file?

**Comment by jeffbean** [comment]
OK with my commits!

**Comment by jeffbean** [comment]
Overall LGTM

File: contrib/ansible/vagrant/README.md (hunk 1 above)
Comment (eparis) [inline contrib/ansible/vagrant/README.md]: I'll add a not here.  vagrant tries to just be smart.

File: contrib/ansible/vagrant/openstack_config.yml (hunk 2 above)
Comment (eparis) [inline contrib/ansible/vagrant/openstack_config.yml]: So I wanted to put real(ish) data here as an example.  Also, I thought the .gitignore trick worked, but it doesn't...

Another option would be to make this file be .gitignore and make a openstack_config.yml.example. And the end user would have to cp the .example to just .yml and then make edits....

I'd still probably leave this data for reference though...

Comment (jeffbean) [inline contrib/ansible/vagrant/openstack_config.yml]: OK just wondering. looks real alight :smile:. that password...

As for the file I haven't played with this so if vagrant tried to pick up this config by default the end-user might not understand why. I would say having a `.example` file extension would make it very clear. Then just call out the copy  in the README.

Comment (eparis) [inline contrib/ansible/vagrant/openstack_config.yml]: I do use crappy passwords, but THAT one isn't one of my crappy passwords   :)

I'll move to the .example version and make the .yml in .gitignore.

**Comment by jayunit100** [comment]
ok w/ my commits also !

**Comment by jayunit100** [comment]
Maybe we can merge this soon as wip and iterate. As of now I think it's useful to people?

**Comment by eparis** [comment]
it's totally G2G from my PoV

**Comment by jayunit100** [comment]
For the record, this is what the right URLs for this looklike.  Would be good to have this in the vagrantfile, just for sanity checking.
```

        os.username     = ENV['OS_USERNAME']
        os.password     = ENV['OS_PASSWORD']
        os.flavor       = "m1.small"
        os.image = "Fedora 22 Cloud Base x86_64 (final)"
        os.openstack_auth_url     = "http://os1-public.osop.rhcloud.com:5000/v2.0/tokens/"
        os.security_groups    = ['default','newgroup']
        os.openstack_compute_url = "http://os1-public.osop.rhcloud.com:8774/v2/abcdabcdabcd"
        os.server_name = name 
        os.tenant_name = "ENG 1234"
        os.keypair_name = "JPeerindex"
        os.region = "OS1Public"
        os.floating_ip_pool   = 'os1_public'
        #os.floating_ip = "auto"
```

**Comment by eparis** [comment]
@jayunit100 I did away with ENV all together.

https://github.com/eparis/kubernetes/blob/total-ansible/contrib/ansible/vagrant/openstack_config.yml.example

**Comment by jayunit100** [comment]
oh duh ok.  well LGTM for sure, ill run quick test now.

**Comment by jayunit100** [comment]
@eparis   i think one minor mod you might want to push before commiting...

```
os_auth_url: "http://os1-public.osop.rhcloud.com:5000/v2.0/"
``` should be 
```
os_auth_url: "http://os1-public.osop.rhcloud.com:5000/v2.0/tokens/"
```
with that change, vagrant up -> spins up VMS+ ansible provisioning  fully automated.  

So definetly ready to merge either way, but suggest updating the URL for os_auth_url

**Comment by eparis** [comment]
porque?  I didn't seem to need the  `tokens/` and it wasn't included anywhere on the security page in the openstack instance I looked at. (although admittedly I agree it works)   Did you have a failure without the `tokens/`?

**Comment by jayunit100** [comment]
Yes, iirc I had to add tokens/.
 I guess we can settle for a comment above that line with the alternate tokens/ endpoint until we work out the exact logic

**Comment by jeffbean** [comment]
@eparis Anything more we need here to merge?

**Comment by eparis** [comment]
I do not believe the jenkin's failure.  This stay entirely inside contrib/ansible.  If jenkins broke, it was a flake.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by eparis** [comment]
@k8s-bot test this please

**Comment by mikedanese** [comment]
Change was isolated to contrib/ansible/ and passing mungedocs and boilerplate checks

**Comment by jayunit100** [comment]
Great! Now we have upstream support for ansible on openstack as well as local vms. Thanks for finishing this of eric and Jeff.

**Pull request merged**